R/module_nested_tabs.R

```diff
@@ -247,24 +254,61 @@ srv_teal_module.teal_module <- function(id,
     filtered_teal_data <- eventReactive(trigger_data(), {
       .make_teal_data(modules, data = data_rv(), datasets = datasets(), datanames = active_datanames())
     })
-    srv_data_summary("data_summary", filtered_teal_data)
+
+    # Needs to be wrapped in reactive to avoid initialization issues when there
+    # filtered data.
+    transformed_teal_data <- reactive({
```

Comment:
I conclude from Friday's comments that there was/is some reactivity problem - putting module initialization in a reactive shouldn't be the solution. This causes:
- evaluating of unnecessary code
- reinstantiating all observers. It burdens shiny-js socket with unnecessary calls
- not sure what it means exactly to overwrite reactive instances in shiny.